hackathon_site/dashboard/frontend/src/components/inventory/InventoryFilter/InventoryFilter.test.js

```diff
@@ -1,60 +1,134 @@
             />
         );
         for (let c of inventoryCategories) {
             expect(queryByText(c.name)).toBeTruthy();
         }
-        for (let o of orderBy) {
-            expect(queryByText(o)).toBeTruthy();
+        for (let o of orderByOptions) {
+            expect(queryByText(o.label)).toBeTruthy();
         }
         expect(getByText("Order by")).toBeInTheDocument();
         expect(getByText("Availability")).toBeInTheDocument();
         expect(getByText("Categories")).toBeInTheDocument();
     });
+
+    it("Submits the form, the clears it, and recieves the expected values", async () => {
+        const handleSubmitSpy = jest.fn();
+        const handleResetSpy = jest.fn();
+        let orderBy = "A-Z";
+        let inStock = true;
+        let inventoryCategories = ["MCU"];
+
+        const { findByLabelText, findByText } = render(
+            <EnhancedInventoryFilter
+                handleSubmit={handleSubmitSpy}
+                handleReset={handleResetSpy}
+                isApplyLoading={false}
+                isClearLoading={false}
+            />
+        );
+
+        const orderByInput = await findByLabelText("A-Z");
+        const inStockInput = await findByLabelText("In stock");
+        const inventoryCategoriesInput = await findByLabelText("MCU");
+        const buttonSubmit = await findByText("Apply");
+        const buttonClear = await findByText("Clear all");
+
+        // Select checkboxes/radio buttons and submit
+        fireEvent.click(orderByInput);
+        fireEvent.click(inStockInput);
+        fireEvent.click(inventoryCategoriesInput);
+        fireEvent.click(buttonSubmit);
+
+        await waitFor(() => {
+            expect(handleSubmitSpy).toHaveBeenCalledWith({
+                orderBy,
+                inStock,
+                inventoryCategories,
+            });
+        });
+
+        orderBy = "Default";
+        inStock = false;
+        inventoryCategories = [];
+
+        // Clear form
+        fireEvent.click(buttonClear);
+
+        await waitFor(() => {
+            expect(handleResetSpy).toHaveBeenCalledWith({
+                orderBy,
+                inStock,
+                inventoryCategories,
+            });
+        });
+    });
+
+    it("Looks for loader on Apply button", () => {
+        const { getByTestId, queryByTestId } = render(
+            <EnhancedInventoryFilter isApplyLoading={true} isClearLoading={false} />
+        );
+
+        expect(getByTestId("circularProgressApply")).toBeInTheDocument();
+        expect(queryByTestId("circularProgressClear")).not.toBeInTheDocument();
+    });
+
+    it("Looks for loader on Clear all button", () => {
+        const { getByTestId, queryByTestId } = render(
+            <EnhancedInventoryFilter isApplyLoading={false} isClearLoading={true} />
+        );
+
+        expect(queryByTestId("circularProgressApply")).not.toBeInTheDocument();
+        expect(getByTestId("circularProgressClear")).toBeInTheDocument();
+    });
```

Comment:
Can you also test that the apply/clear buttons are disabled when either of them is loading? The easiest thing would be to just add checks for those buttons being disabled into the two loading tests above